### ISSUE TITLE
can't compile because of function 'findClients'

### ISSUE THREAD

**Issue by redwill17** [codic12/worm#91]
trying to compile with nimble gives me this error

```
/home/javu/build/worm/src/wm.nim(172, 6) Error: 'findClient' can have side effects
> /home/javu/build/worm/src/wm.nim(178, 18) Hint: 'findClient' calls routine via hidden pointer indirection

       Tip: 30 messages have been suppressed, use --verbose to show them.
nimble.nim(229)          buildFromDir

    Error:  Build failed for the package: worm
```

i know that side effect functions are called procs, so using that keyword for findClients i was able to compile the program, I haven't tested the binary yet and i'll make an update when I do, but how odd is this error? i'm using void linux and used xbps to install nimble, maybe that's part of the problem

**Comment by Atrevix** [comment]
I had the same problem, thanks for the fix.

**Comment by codic12** [comment]
Looks like nim 2.0 has changed some things. Unfortunately I haven't had time to work on worm for a while. Could you try to make a PR for your change? Thanks!

**Comment by Peter2121** [comment]
@redwill17 

> using that keyword for findClients i was able to compile the program

Could you, please, put the exact syntax anywhere?

**Comment by codic12** [comment]
closed by #91 (and so sorry for the late reply)
